test/test_muscles.cpp

```diff
@@ -304,6 +306,144 @@ TEST(hillType, copy)
+
+    }
+    {
+        biorbd::Model model(modelPathForMuscleForce);
+        biorbd::muscles::HillThelenType hillThelenType(
+            model.muscleGroup(muscleGroupForHillThelenType).muscle(
+                muscleForHillThelenType));
+        biorbd::rigidbody::GeneralizedCoordinates Q(model);
+        biorbd::rigidbody::GeneralizedVelocity qDot(model);
+        Q = Q.setOnes() / 10;
+        qDot = qDot.setOnes() / 10;
+        model.updateMuscles(Q, 2);
+        hillThelenType.updateOrientations(model, Q, qDot);
+        static double activationEmgForHillTypeTest(1.0);
+        biorbd::muscles::StateDynamics emg(0, activationEmgForHillTypeTest);
+
+        EXPECT_NEAR(hillThelenType.FlCE(emg), 0.67988981401208015, requiredPrecision);
+        EXPECT_NEAR(hillThelenType.FlPE(), 0, requiredPrecision);
+        EXPECT_NEAR(hillThelenType.FvCE(), 1.000886825333013, requiredPrecision);
+        EXPECT_NEAR(hillThelenType.damping(), 0.00019534599393617336, requiredPrecision);
+        EXPECT_NEAR(hillThelenType.force(emg), 424.95358302550062, requiredPrecision);
+    }
+}
+
+TEST(hillThelenType, copy)
+{
+    {
+        biorbd::Model model(modelPathForMuscleForce);
+        biorbd::muscles::HillThelenType hillThelenType(
+            model.muscleGroup(muscleGroupForHillThelenType).muscle(
+                muscleForHillThelenType));
+        biorbd::rigidbody::GeneralizedCoordinates Q(model);
+        Q = Q.setOnes() / 10;
+
+        biorbd::muscles::HillThelenType shallowCopy(hillThelenType);
+        biorbd::muscles::HillThelenType deepCopyNow(hillThelenType.DeepCopy());
+        biorbd::muscles::HillThelenType deepCopyLater;
+        deepCopyLater.DeepCopy(hillThelenType);
+
+        biorbd::utils::String originalName(hillThelenType.name());
+        EXPECT_STREQ(shallowCopy.name().c_str(), originalName.c_str());
+        EXPECT_STREQ(deepCopyNow.name().c_str(), originalName.c_str());
+        EXPECT_STREQ(deepCopyLater.name().c_str(), originalName.c_str());
+
+        biorbd::utils::String newName("MyNewMuscleName");
+        hillThelenType.setName(newName);
+        EXPECT_STREQ(hillThelenType.name().c_str(), newName.c_str());
+        EXPECT_STREQ(shallowCopy.name().c_str(), newName.c_str());
+        EXPECT_STREQ(deepCopyNow.name().c_str(), originalName.c_str());
+        EXPECT_STREQ(deepCopyLater.name().c_str(), originalName.c_str());
+    }
+    {
+        biorbd::Model model(modelPathForMuscleForce);
+        biorbd::muscles::HillThelenType hillThelenType(
+            model.muscleGroup(muscleGroupForHillThelenType).muscle(
+                muscleForHillThelenType));
+
+        biorbd::muscles::HillThelenType shallowCopy(hillThelenType);
+        biorbd::muscles::HillThelenType deepCopyNow(hillThelenType.DeepCopy());
+        biorbd::muscles::HillThelenType deepCopyLater;
+        deepCopyLater.DeepCopy(hillThelenType);
+
+        double pennationAngleOriginal(hillThelenType.characteristics().pennationAngle());
```

Comment:
Same here, the value of the variable should be a hard-coded value and not the one returned from the muscle